doc/guides/tfm/testsuites.rst

```diff
@@ -0,0 +1,51 @@
+Test Suites
+###########
+
+TF-M includes two sets of test suites:
+
+* tf-m-tests - Standard regression tests, and test to verify the NS interface
+* psa-arch-tests - Test suites for specific APIs (secure storage, etc.)
+
+These test suites can be run from Zephyr via an appropriate sample application
+in the samples/tfm_integration folder.
+
+PSA Regression Tests
+********************
+
+The regression test suite can be run via:
+
+   .. code-block:: bash
+
+     $ west build samples/tfm_integration/tfm_regression_test/ \
+       -p -b mps2_an521_ns -t run
+
+The output will appear, in this case, in the console. When running this on
+actual hardware, it will be output to an appropriate UART peripheral, etc.
+
+This sample is useful if you require PSA certification for your board.
+The test artifacts are required as part of the certification process to
+verify PSA API compliance with the NS/S interface.
+
+PSA Arch Tests
+**************
+
+The PSA Arch Test suite contains a number of test suites that can be run.
+Only one of these suites can be run at a time, with the test to run set
+via one of the following kconfig options:
+
+* ``CONFIG_TFM_PSA_TEST_CRYPTO``
+* ``CONFIG_TFM_PSA_TEST_PROTECTED_STORAGE``
+* ``CONFIG_TFM_PSA_TEST_INTERNAL_TRUSTED_STORAGE``
+* ``CONFIG_TFM_PSA_TEST_STORAGE``
+* ``CONFIG_TFM_PSA_TEST_INITIAL_ATTESTATION``
+
+You can indicate the desired test suite at build time via:
+
+   .. code-block:: bash
+
+     $ west build samples/tfm_integration/tfm_psa_test/ \
+       -p -b mps2_an521_ns -t run -- \
+       -DCONFIG_TFM_PSA_TEST_STORAGE=y
+
+The output will appear, in this case, in the console. When running this on
+actual hardware, it will be output to an appropriate UART peripheral, etc.
```

Comment:
If I were running tfm_psa_test I would expect to find the documentation in the sample's documentation, isn't this documentation a bit redundant / misplaced ?

Reply:
I was simply hoping to bring the samples and test suites to people's attention since many people might not be aware of the distinction or why one is useful versus the other. I agree this may not be a relevant sentence in this case and can update with a better description of what the two suites do.

Reply:
I thought maybe you were unaware of the sample documentation.

If this is the purpose then this is ok.

Reply:
But I think we should focus on the "why" you would want to run the samples here.

And the "how" in the sample documentation.

Reply:
I pushed an update out, although I still left the sample west command since I think it's useful to list to just run them quickly, even if it also exists in the sample readme.